### DISCUSSION TITLE
How to disable sampling

### DISCUSSION THREAD

**Discussion by pdorrestijn** [uber-go/zap#1212] [Q&A] [answered]
On stackoverflow i see some references towards zap code on how to disable sampling.
However, this all refers to using a zap.Config type while I'm using zap with lumberjack and I can't seem to find a way to disable sampling.
I tried to investigate how to disable sampling, but I can't find it.
Sampling is really undesirable for me, please help me out.
This is how I managed to build the zap logger using a lumberjack writer which is added to zap using zapcore.AddSync:
`
core = zapcore.NewCore(getEncoder(), getLogWriter(_globalLogConfiguration), _level)

_logger = zap.New(core)
`

**Comment by abhinav** [comment, marked as answer]
If you're constructing a logger like that code sample -- with `zapcore.NewCore` followed by `zap.New`,
sampling is already disabled.

To *enable* sampling on top of that, you'd have to use [zapcore.NewSamplerWithOptions](https://pkg.go.dev/go.uber.org/zap/zapcore#NewSamplerWithOptions).

```go
core = zapcore.NewSamplerWithOptions(...)
```

But as-is, without any additional wrapping, sampling is already disabled.

The other way you could've been constructing the logger is with [`zap.Config`](https://pkg.go.dev/go.uber.org/zap#Config).[`Build()`](https://pkg.go.dev/go.uber.org/zap#Config.Build),
in which case, to disable sampling you'd have to set [`Config.Sampling`](https://pkg.go.dev/go.uber.org/zap#Config.Sampling) to nil.

**Comment by pdorrestijn** [comment]
Nevermind :)
I found the issue, zap rocks!
